src/test/java/bio/terra/common/exception/SamApiExceptionTest.java

```diff
@@ -0,0 +1,105 @@
+package bio.terra.common.exception;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.HttpStatusCodes;
+import java.util.HashMap;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.model.ErrorReport;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Test for {@link SamApiException} */
+@Tag("unit")
+public class SamApiExceptionTest {
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  public void extractsMessageFromSamTopLevel() {
+    String message = "Top level message";
+    ApiException apiException = new ApiException("Top level message");
+    SamApiException samApiException = SamApiException.createSamApiException(apiException);
+    assertEquals("Error from Sam: " + message, samApiException.getMessage());
+  }
+
+  @Test
+  public void extractsMessageBuriedInSam() {
+    String message = "Buried message";
+    ErrorReport errorReport = new ErrorReport().message(message);
+    try {
+      String errorReportString = objectMapper.writeValueAsString(errorReport);
+      ApiException apiException =
+          new ApiException(
+              "", HttpStatusCodes.STATUS_CODE_BAD_REQUEST, new HashMap<>(), errorReportString);
+      SamApiException samApiException = SamApiException.createSamApiException(apiException);
+      assertEquals("Error from Sam: " + message, samApiException.getMessage());
+    } catch (JsonProcessingException e) {
+      // FAIL
```

Comment:
I don't think this fails. Fine to add `throws Exception` to an `@Test` method and Junit will handle failing the `@Test` if some exception is thrown.

Reply:
Ah my bad

Reply:
So I think you should remove the catch of JsonProcessingException, yes?

Reply:
Should be gone now